### DISCUSSION TITLE
Generic support for data class-based bindings

### DISCUSSION THREAD

**Discussion by shreyabatra4** [Azure/azure-functions-python-worker#1326] [Ideas]
Curious to hear whether generic support for data class-based bindings would improve your experience developing functions.

From [this issue](https://github.com/Azure/azure-functions-python-worker/issues/416):

One possible way to implement generic handling of bindings is to allow users to annotate function parameters with a [dataclass](https://docs.python.org/3/library/dataclasses.html). The argument data is then treated as a conforming JSON object.
  
  For example:
  
```
  import dataclasses
  
  @dataclasses.dataclass
  class MyType:
      id: int
      body: str
  
  def func(arg: MyType) -> str:
      return arg.body
```

  In function.json such bindings would then have to specify dataType: json.

**Comment by danielniccoli** [comment]
I would suggest to treat Pydantic as a first-class citizen, instead of developing a custom implementation that's inferior to an already widely used library, that's created for the exact same purpose. In addition to that, also treat LiteStar and FastAPI as a first-class citizen for OpenAPI support, which in turn treats Pydantic as an integral part of that.
